hooks/boot/01-kafo-hook-extensions.rb

```diff
@@ -20,7 +20,11 @@ def ensure_package(package, state = 'installed')
     end
 
     bin_path = Kafo::PuppetCommand.search_puppet_path('puppet')
-    execute("#{bin_path} resource package #{package} ensure=#{state}")
+    response = `#{bin_path} resource package #{package} ensure=#{state}`
+
+    if state == 'installed' && response.include?('purged')
```

Comment:
Puppet has an exit code of 0 if nothing changed and 2 when there was a change. It would be better to catch that and state it couldn't install the package.